### PR TITLE
dashboard view components: moving more logic to ruby

### LINKED ISSUE #2079: Dashboard view components: move more logic into ruby
per JLitt's comment on PR https://github.com/sul-dlss/preservation_catalog/pull/2076

e.g. in audit_information_component we have

```erb
      <td class="text-end<%= ' table-warning' if moab_audits_older_than_threshold? %>">Moab audits older than <%= moab_audit_age_threshold %>:<hr/><%= num_moab_audits_older_than_threshold %></td>
      <td class="text-end<%= ' table-danger' if CompleteMoab.invalid_checksum.count != 0 %>"><%= CompleteMoab.invalid_checksum.count %></td>
      <td class="text-center<%= ' table-warning' if num_expired_checksum_validation != 0 %>"><%= num_expired_checksum_validation %><br/><br/>(passing checks expire 90 days after they are run)</td>
```

should be 

```erb
      <td class="text-end<%= ' table-warning' if moab_audits_older_than_threshold? %>">Moab audits older than <%= moab_audit_age_threshold %>:<hr/><%= num_moab_audits_older_than_threshold %></td>
      <td class="text-end<%= ' table-danger' if moabs_have_invalid_checksum? %>"><%= CompleteMoab.invalid_checksum.count %></td>
      <td class="text-center<%= ' table-warning' if expired_checksum_validations? %>"><%= num_expired_checksum_validation %><br/><br/>(passing checks expire 90 days after they are run)</td>
```

whether the new methods should be in the existing service classes or in the specific component is up to the implemeter.

### PR DESCRIPTION
**Pull request by ndushay**
## Why was this change made? 🤔

The goal here is to make the erb files more straightforward.

Fixes #2079

## How was this change tested? 🤨

⚡ ⚠ If this change has cross service impact, or if it changes code used internally for cloud replication, ***run [integration test preassembly_image_accessioning_spec.rb](https://github.com/sul-dlss/infrastructure-integration-test/blob/main/spec/features/preassembly_image_accessioning_spec.rb) against stage as it tests preservation***, and/or test in stage environment, in addition to specs. The main classes relevant to replication are `ZipmakerJob`, `PlexerJob`, `*DeliveryJob`, `ResultsRecorderJob`, and `DruidVersionZip`; [see here for overview diagram of replication pipeline](https://github.com/sul-dlss/preservation_catalog/blob/main/app/jobs/README.md).⚡

### CODE REVIEW

File: app/services/dashboard/audit_service.rb [hunk 1]
```diff
@@ -19,7 +19,7 @@ def audits_ok?
       validate_moab_audit_ok? &&
         catalog_to_moab_audit_ok? &&
         moab_to_catalog_audit_ok? &&
-        checksum_validation_audit_ok? &&
+        moab_checksum_validation_audit_ok? &&
```
Comment (ndushay) [inline app/services/dashboard/audit_service.rb:22]: added moab_  to clarify between on prem and cloud checksums

File: app/services/dashboard/catalog_service.rb [hunk 2]
```diff
@@ -117,6 +121,18 @@ def average_version_per_complete_moab
       CompleteMoab.pick(Arel.sql('SUM(version)::numeric/COUNT(id)')).round(2) unless num_complete_moabs.zero?
     end
 
+    def preserved_object_complete_moab_counts_match?
```
Comment (ndushay) [inline app/services/dashboard/catalog_service.rb:124]: yucky name, but couldn't think of a better one that is as clear.

File: app/services/dashboard/catalog_service.rb [hunk 3]
```diff
@@ -117,6 +121,18 @@ def average_version_per_complete_moab
...
+      num_preserved_objects == num_complete_moabs
+    end
+
+    def num_object_versions_preserved_object_complete_moab_match?
```
Comment (ndushay) [inline app/services/dashboard/catalog_service.rb:128]: yucky name, but couldn't think of a better one that is as clear.

File: app/services/dashboard/catalog_service.rb [hunk 4]
```diff
@@ -117,6 +121,18 @@ def average_version_per_complete_moab
...
+      num_object_versions_per_preserved_object == num_object_versions_per_complete_moab
+    end
+
+    def highest_version_preserved_object_complete_moab_match?
```
Comment (ndushay) [inline app/services/dashboard/catalog_service.rb:132]: yucky name, but couldn't think of a better one that is as clear.

File: app/services/dashboard/replication_service.rb [hunk 5]
```diff
@@ -39,5 +43,17 @@ def zip_parts_total_size
...
+      ZipPart.unreplicated.count.positive?
+    end
+
+    def zip_parts_not_found?
+      ZipPart.not_found.count.positive?
+    end
+
+    def zip_parts_replicated_checksum_mismatch?
+      ZipPart.replicated_checksum_mismatch.count.positive?
+    end
```
Comment (ndushay) [inline app/services/dashboard/replication_service.rb:57]: I *think* these make the erb more readable

File: app/services/dashboard/moab_on_storage_service.rb [hunk 6]
```diff
@@ -121,6 +125,18 @@ def average_version_per_complete_moab
       CompleteMoab.pick(Arel.sql('SUM(version)::numeric/COUNT(id)')).round(2) unless num_complete_moabs.zero?
     end
 
+    def preserved_object_complete_moab_counts_match?
```
Comment (ndushay) [inline app/services/dashboard/moab_on_storage_service.rb:128]: yucky name, but couldn't think of a better one that is as clear.

File: app/services/dashboard/moab_on_storage_service.rb [hunk 7]
```diff
@@ -121,6 +125,18 @@ def average_version_per_complete_moab
...
+      num_preserved_objects == num_complete_moabs
+    end
+
+    def num_object_versions_preserved_object_complete_moab_match?
```
Comment (ndushay) [inline app/services/dashboard/moab_on_storage_service.rb:132]: yucky name, but couldn't think of a better one that is as clear.

File: app/services/dashboard/moab_on_storage_service.rb [hunk 8]
```diff
@@ -121,6 +125,18 @@ def average_version_per_complete_moab
...
+      num_object_versions_per_preserved_object == num_object_versions_per_complete_moab
+    end
+
+    def highest_version_preserved_object_complete_moab_match?
```
Comment (ndushay) [inline app/services/dashboard/moab_on_storage_service.rb:136]: yucky name, but couldn't think of a better one that is as clear.

**Review by aaron-collier — APPROVED** [review]

**Pull request merged**